From 4850a5fac4b9aeb607a4ff72cf65ebaee72066bc Mon Sep 17 00:00:00 2001
From: hariprasad <hkelam@marvell.com>
Date: Fri, 9 Aug 2019 15:03:49 +0530
Subject: [PATCH 0256/1921] octeontx2-pf: Add validation in ntuple delete flow

Add proper validation to handle below cases
 *mcam rules are truncated
 *user calls ntuple off without adding any

Change-Id: Ic1ab6b2ac72c7fb33b22b1cecd2fba240bf0743c
Signed-off-by: hariprasad <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/14087
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_flows.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_flows.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_flows.c
index 95cf8d7400ce..b94372977be2 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_flows.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_flows.c
@@ -431,6 +431,9 @@ int otx2_destroy_ntuple_flows(struct otx2_nic *pfvf)
 	struct otx2_flow *iter, *tmp;
 	int err;
 
+	if (!pfvf->entries_alloc)
+		return 0;
+
 	otx2_mbox_lock(&pfvf->mbox);
 	req = otx2_mbox_alloc_msg_npc_delete_flow(&pfvf->mbox);
 	if (!req) {
@@ -440,7 +443,7 @@ int otx2_destroy_ntuple_flows(struct otx2_nic *pfvf)
 
 	req->start = pfvf->entry_list[NTUPLE_OFFSET];
 	req->end   = pfvf->entry_list[NTUPLE_OFFSET +
-				      OTX2_MAX_NTUPLE_FLOWS - 1];
+				      pfvf->ntuple_max_flows - 1];
 	err = otx2_sync_mbox_msg(&pfvf->mbox);
 	otx2_mbox_unlock(&pfvf->mbox);
 
-- 
2.31.1

